suites/apis/googleServiceAccountRemovalTest.js

```diff
@@ -0,0 +1,471 @@
+
+  chai.expect(unlockRes, 'Could not unlock project').to.be.true;
+}).retry(2);
+
+
+// TODO: enable. this test fails until the jenkins envs are setup with the new monitoring SA
+xScenario('SA removal job test: monitor SA does not have access @reqGoogle', async (fence, users, google, files) => {
+  // test invalid SA because the monitor does not have access
+
+  // Setup
+  const googleProject = fence.props.googleProjectDynamic;
+  let getMonitorRes = await fence.do.getGoogleServiceAccountMonitor(users.user0);
+  fence.ask.assertStatusCode(getMonitorRes, 200, 'Could not get SA monitor info');
+  let monitorEmail = getMonitorRes.body.service_account_email;
+
+  await fence.complete.forceLinkGoogleAcct(users.user0, googleProject.owner);
+
+  // Register account
+  registerRes = await fence.do.registerGoogleServiceAccount(
+    users.user0,
+    googleProject,
+    ['QA']
+  );
+  fence.ask.responsesEqual(registerRes, fence.props.resRegisterServiceAccountSuccess);
+
+  // Lock the project
+  lockRes = await google.lockGoogleProject(googleProject);
+  chai.expect(lockRes, 'Could not lock project').to.be.true;
+
+  // Remove monitor's access
+  console.log(`removing monitoring access from SA ${monitorEmail}`);
+  const monitorRoles = [ // IAM access needed by the monitor
+    'roles/resourcemanager.projectIamAdmin',
+    'roles/editor'
+  ];
+  for (var role of monitorRoles) {
+    await google.removeUserRole(
+      googleProject.id,
+      {
+        role,
+        members: [`serviceAccount:${monitorEmail}`]
+      }
+    );
+  }
+
+  // the role update can take a bit of time to propagate
+  /**
+   * return true if the job detected an invalid project, false otherwise
+   */
+  const isInvalidProjectDetected = async function() {
+    try {
+      let jobRes = checkAndCleanSA(); // run clean up job
+      fence.ask.detected_invalid_google_project(jobRes, fence.props.monitorSAJobLog.noMonitorAccess); // check results
+      return true;
+    }
+    catch {
+      // assert failed: no problem detected by the clean up job yet
+      console.log('No invalid project detected by the job yet: rerunning');
+      return false;
+    }
+  };
+  const startWait = 10; // start by waiting 10 secs
+  const timeout = 120; // max number of seconds to wait
+  detected_invalid_google_project = true;
+  try {
+    await apiUtil.smartWait(isInvalidProjectDetected, [], timeout, '', startWait);
+  }
+  catch {
+    detected_invalid_google_project = false;
+  }
+
+  // Try to access data
+  let [pathToKeyFile, keyFullName] = await google.createServiceAccountKeyFile(googleProject);
+  user0AccessQARes = await google.getFileFromBucket(
+    fence.props.googleBucketInfo.QA.googleProjectId,
+    pathToKeyFile,
+    fence.props.googleBucketInfo.QA.bucketId,
+    fence.props.googleBucketInfo.QA.fileName
+  );
+
+  // Clean up
+  console.log('cleaning up');
+
+  // Add monitor's access back
+  for (var role of monitorRoles) {
+    await google.updateUserRole(
```

Comment:
what if this fails? we should probably check the response and provide help if it fails.. b/c someone will have to manually re-add it, right? probably safe to try and add it add the beginning and catch any conflict errors and ignore/continue. that way hopefully no one has to manually touch these projects even if this fails

Reply:
i added a second role update in the `after suite` to make sure it gets done - we can't just add it everywhere because we need to lock the project to updates roles, and that's messy
i will also move this test to run last in the suite, so that a failure to update the roles doesn't make the following tests fail